### ISSUE TITLE
mistake in help text of ?data.table for with?

### ISSUE THREAD

**Issue by dracodoc** [Rdatatable/data.table#4955]
I'm revisiting basics by reading help of `?data.table`, then I found this part

> - with
> with=FALSE is not necessary anymore to select columns dynamically. Note that x[, cols] is equivalent to x[, ..cols] and to x[, cols, with=FALSE] and to x[, .SD, .SDcols=cols].

`x[, cols] is equivalent to x[, ..cols]` is a surprise for me and a simple test contradict with that.

```r
> dt <- data.table(mtcars)
> test <- "mpg"
> dt[, test]
Error in `[.data.table`(dt, , test) : 
  j (the 2nd argument inside [...]) is a single symbol but column name 'test' is not found. Perhaps you intended DT[, ..test]. This difference to data.frame is deliberate and explained in FAQ 1.1.
> dt[, ..test]
     mpg
 1: 21.0
 2: 21.0
```

Is there some typo in this part of help? I'm using data.table 1.14.0

**Comment by dvg-p4** [comment]
The FAQ also implies this should work:
> Before v1.9.8 released Nov 2016, DT[,5] used to just return 5. The thinking was that we could more simply teach one fact that the parts inside DT[...] get evaluated within the frame of DT always (they see column names as if they are variables). And 5 evaluates to 5 so that behaviour was consistent with the single rule. We asked you to go through an extra deliberate hurdle DT[,5,with=FALSE] if you really wanted to select a column by name or number. Going forward from Nov 2016, you don’t need to use with=FALSE and we’ll see how greater consistency with data.frame in this regard will help or hinder both new and long-time users.

**Comment by jangorecki** [comment]
Its get even more tricky when name of variable is the same as name of the column in DT
```r
mpg = "hp"
dt[, mpg]
```
ultimate solution is to provide an alternative interface that does not have NSE, like the new `env` arg.

**Comment by Mukulyadav2004** [comment]
Hi @MichaelChirico @jangorecki 
What I'm understanding here is when we write `DT[, cols]`, probably `data.table` thinks we want a column literally named 'cols', but there's no column called "cols" in the data so it gives an error.
And when we write `DT[, ..cols]`, data.table maybe thought of it as - ".." means look up what's stored in the variable 'cols' so it finds `cols` contains `c("mpg", "hp")`thus it selects those columns.

So can we add - when cols is a literal character vector, x[, cols] is equivalent to x[, ..cols]
Is my understanding correct here?

**Comment by MichaelChirico** [comment]
Yes, though I think what's intended is `dt[, cols]` _for data.frames_ is equivalent to `dt[, ..cols]`, ... _for data.tables_. The sentence is addressing the historical need for `with=FALSE` to induce the behavior for those more comfortable with data.frame rules: instead of `dt[, cols]` on data.frame, use `dt[, cols, with=FALSE]` in data.table would have been the "old" recommendation.

I think between my historical context above and your clarification, we have enough to update the docs for clarity.

**Comment by MichaelChirico** [comment]
Note that OP is also omitting a crucial part of the existing description (added in 2018, before this issue: c72a977b269281e3b140f9408508746e5b04e77f):

https://github.com/Rdatatable/data.table/blob/bfa049ca5f3f15b7f8c68013d3b42d582594f840/man/data.table.Rd#L122
